### DISCUSSION TITLE
Require some information for caffeine cache.

### DISCUSSION THREAD

**Discussion by kamalnigam** [ben-manes/caffeine#1784] [General]
Hey, 

- I would like to use caffeine in our project.  We need to use only Java 1.8 in our project. Does caffeine have long-term support with Java 1.8?  
- Can we set the max size (in bytes) of the cache in caffeine? Do we have only the below implementation?
    CacheBuilder.weigher(Weigher) and CacheBuilder. Maximum eight (long).
- How do you reset cache key expiry in caffeine?

**Comment by ben-manes** [comment]
1. Yes, in that critical bugs will be fixed but beyond that improvements will not be back ported (see https://openjdk.org/jeps/14). Be warned that Java 8 has reached end of life for public updates.
2. The weigher can be used for byte sizing. You have to define your own weigher since Java object sizes is a tricky (https://openjdk.org/jeps/8249196).
3. You might use expireAfter(Expiry) or the Cache.policy() methods if you need advanced functionality.
